app/src/main/java/com/example/henzoshimada/feeltrip/ElasticSearchController.java

```diff
@@ -64,46 +66,84 @@ protected Void doInBackground(Mood ... moods ) {
 
         @Override
         protected Void doInBackground(Mood ... moods ) {
+            if(android.os.Debug.isDebuggerConnected())
+                android.os.Debug.waitForDebugger();
             verifySettings();
 
             for (Mood mood : moods) {
                 if (!mood.isChanged()){
                     return null;
                 }
-                String moodId = mood.getId();
-                //Index index = new Index.Builder(mood).index(groupIndex).type("mood").id(moodId).build();
-                String query = "{\n";
-                Boolean changed;
+                String moodId = mood.getId(); //TODO: WHat happens if there's no valid ID? Shouldn't happen, but still
+                String query = "{\"doc\" : {";
                 // find out what fields have been changed and build query accordingly
-                for (int i = 0; i < 7; i++){
-                    changed = mood.getStateByIndex(i);
-                    if (changed){
-                        switch (i){
+                int notDone = 0;
+                for (int i = 0; i < 8; i++) { //find out how many fields were changed
+                    if (mood.getStateByIndex(i)) {
+                        notDone += 1;
+                    }
+                }
+                for (int i = 0; i < 8; i++) {
+                    if (mood.getStateByIndex(i)) {
+                        notDone -= 1;
+                        switch (i) {
                             case 0:
```

Comment:
Here we build the query, according to what has been edited and only what has been edited. The notDone field tells us whether a comma is necessary.